streams/src/main/java/org/apache/kafka/streams/kstream/internals/metrics/Sensors.java

```diff
@@ -38,13 +45,89 @@ public static Sensor lateRecordDropSensor(final InternalProcessorContext context
         );
         StreamsMetricsImpl.addInvocationRateAndCount(
             sensor,
-            "stream-processor-node-metrics",
+            PROCESSOR_NODE_METRICS_GROUP,
             metrics.tagMap("task-id", context.taskId().toString(), "processor-node-id", context.currentNode().name()),
             "late-record-drop"
         );
         return sensor;
     }
 
+    public static Sensor suppressionSensor(final InternalProcessorContext context) {
+        final StreamsMetricsImpl metrics = context.metrics();
+        final String operation = "intermediate-result-suppression";
+
+        final Sensor sensor = metrics.nodeLevelSensor(
+            context.taskId().toString(),
+            context.currentNode().name(),
+            operation,
+            Sensor.RecordingLevel.DEBUG
+        );
+
+        final Map<String, String> tags = metrics.tagMap(
+            "task-id", context.taskId().toString(),
+            "processor-node-id", context.currentNode().name()
+        );
+
+        sensor.add(
+            new MetricName(
+                operation + "-rate",
+                PROCESSOR_NODE_METRICS_GROUP,
+                "The average number of occurrence of " + operation + " operation per second.",
+                tags
+            ),
+            new Rate(TimeUnit.SECONDS, new Sum())
+        );
+        sensor.add(
+            new MetricName(
+                operation + "-total",
+                PROCESSOR_NODE_METRICS_GROUP,
+                "The total number of occurrence of " + operation + " operations.",
+                tags
+            ),
+            new Total()
+        );
+
+        return sensor;
+    }
+
+    public static Sensor suppressionEmitSensor(final InternalProcessorContext context) {
+        final StreamsMetricsImpl metrics = context.metrics();
+        final String operation = "suppression-emit";
+
+        final Sensor sensor = metrics.nodeLevelSensor(
+            context.taskId().toString(),
+            context.currentNode().name(),
+            operation,
+            Sensor.RecordingLevel.DEBUG
+        );
+
+        final Map<String, String> tags = metrics.tagMap(
+            "task-id", context.taskId().toString(),
+            "processor-node-id", context.currentNode().name()
+        );
+
+        sensor.add(
+            new MetricName(
```

Comment:
seems like the contents of the two `sensor#add` methods are the same as above maybe refactor into two methods?